### PR TITLE
nrf_security: Misc legacy fixes for nRF Connect SDK 1.9.0

### PR DESCRIPTION
**Pull request by frkv**
-Disabling unnecessary auto-select of OBERON_BACKEND for all
 non TF-M builds using legacy APIs
-Enabling removal of prompt for OBERON_BACKEND in case CC3XX_BACKEND
 is enabled
-Enabling DHM for cc3xx (missing APIs in CryptoCell runtime)

ref: NCSDK-13773
ref: NCSDK-13762
ref: NCSDK-13744

Signed-off-by: Frank Audun Kvamtrø <frank.kvamtro@nordicsemi.no>

### CODE REVIEW

**Review by edmont — APPROVED** [review]

**Review by tejlmand — DISMISSED** [review]

**Comment by frkv** [comment]
PR passes long-running tests, ready for final review and merging

**Review by Vge0rge — APPROVED** [review]

File: crypto/nrf_cc310_mbedcrypto/include/mbedtls/rsa_alt.h [hunk 1]
```diff
@@ -23,6 +23,7 @@
 extern "C" {
 #endif
 
+
```
Comment (tejlmand) [inline crypto/nrf_cc310_mbedcrypto/include/mbedtls/rsa_alt.h:26]: why a double empty line here ?

File: crypto/nrf_cc310_mbedcrypto/include/mbedtls/rsa_alt.h [hunk 2]
```diff
@@ -62,6 +63,8 @@ typedef struct
                                       encoding                          */
 #if defined(MBEDTLS_THREADING_C)
     mbedtls_threading_mutex_t mutex;    /*!<  Thread-safety mutex       */
+#else
+    uint8_t dummy[8];
```
Comment (tejlmand) [inline crypto/nrf_cc310_mbedcrypto/include/mbedtls/rsa_alt.h:67]: would be nice with a comment explaining that this is to ensure context size is always the same size, and that `mbedtls_threading_mutex_t` is 8 byte large (I assume)

File: crypto/nrf_cc312_mbedcrypto/include/mbedtls/rsa_alt.h [hunk 3]
```diff
@@ -23,6 +23,7 @@
 extern "C" {
 #endif
 
+
```
Comment (tejlmand) [inline crypto/nrf_cc312_mbedcrypto/include/mbedtls/rsa_alt.h:26]: why a double empty line here ?

File: crypto/nrf_cc312_mbedcrypto/include/mbedtls/rsa_alt.h [hunk 4]
```diff
@@ -62,6 +63,8 @@ typedef struct
                                       encoding                          */
 #if defined(MBEDTLS_THREADING_C)
     mbedtls_threading_mutex_t mutex;    /*!<  Thread-safety mutex       */
+#else
+    uint8_t dummy[8];
```
Comment (tejlmand) [inline crypto/nrf_cc312_mbedcrypto/include/mbedtls/rsa_alt.h:67]: also here a comment would be nice.

**Review by tejlmand — COMMENTED** [review]
some minor nits, nothing blocking.

Rest looks good.

File: crypto/nrf_cc312_mbedcrypto/include/mbedtls/rsa_alt.h (hunk 3 above)
Comment (frkv) [reply crypto/nrf_cc312_mbedcrypto/include/mbedtls/rsa_alt.h:26]: Fixed, part of next force-push

**Review by tejlmand — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

**Review by torsteingrindvik — APPROVED** [review]
Verified via related manifest PR.

**Pull request merged**